### PR TITLE
feat: func isEmptyValue support time.Time

### PR DESCRIPTION
**Pull request by gh73962**
This PR is based on the solution discussed in the  #3208 issue.

### CODE REVIEW

**Comment by ndyakov** [comment]
@gh73962 Thank you for the contribution. The test that you have added would have passed with or without the addition you did. Can we check what will be read so we verify the data. Maybe a raw read would work, since if we unmarshal the response to the object it would be the same as before, won't it?

**Comment by gh73962** [comment]
> @gh73962 Thank you for the contribution. The test that you have added would have passed with or without the addition you did. Can we check what will be read so we verify the data. Maybe a raw read would work, since if we unmarshal the response to the object it would be the same as before, won't it?

Of course, I don't have the capability to perform this kind of test on my side. Changes like this should have comprehensive tests to ensure code quality.

**Comment by ndyakov** [comment]
@gh73962 would you like me to help you add those tests? The CI will run a redis server, you should be able to execute code agains it in the tests (you can check the existing tests for Set and Get). Let me know if I can help you somehow.

**Comment by gh73962** [comment]
> @gh73962 would you like me to help you add those tests? The CI will run a redis server, you should be able to execute code agains it in the tests (you can check the existing tests for Set and Get). Let me know if I can help you somehow.

Of course

**Comment by ndyakov** [comment]
@gh73962 any concrete questions on how to write tests? will you be able to do this on your own?

**Comment by gh73962** [comment]
In my first commit of the `commands_test.go` file, the added test covers a scenario that our team frequently encounters.I need you guide, What additional test cases should I include?

**Comment by ndyakov** [comment]
@gh73962 lets add tests for the following scenarious:
- Add a test with custom struct in the hash that wants to hold a zero value (for example `type RequestsSetting struct { Limit int }` and the limit set to zero. Check if the struct is parsed correctly when the data is requested from redis.
- Add a test which validates that the data that was received from redis is what was written to redis. For example add a test where you set the `time.Time` value and validate that the same is parsed later when reading the data.

**Comment by gh73962** [comment]
> @gh73962 lets add tests for the following scenarios:
> 
>     * Add a test with custom struct in the hash that wants to hold a zero value (for example `type RequestsSetting struct { Limit int }` and the limit set to zero. Check if the struct is parsed correctly when the data is requested from redis.
> 
>     * Add a test which validates that the data that was received from redis is what was written to redis. For example add a test where you set the `time.Time` value and validate that the same is parsed later when reading the data.

add to `commands_test.go` `It("should HSet", func()` ?

**Comment by ndyakov** [comment]
@gh73962 yes.

**Comment by ndyakov** [comment]
@gh73962 can we try with a child struct? Would you like me to write the tests or you would do it?

**Review by ndyakov — CHANGES_REQUESTED** [review]
I don't find the testing sufficient. Please check the discussion so far.

**Comment by gh73962** [comment]
@ndyakov I was sick a few days ago, and I'll start improving these tasks tomorrow.

**Comment by gh73962** [comment]
> @gh73962 can we try with a child struct? Would you like me to write the tests or you would do it?

I can test it out and check the impact on the child struct. Should I write the tests, or will you?

**Comment by gh73962** [comment]
@ndyakov pls review and hset.Scan not support struct
图片

File: commands.go [hunk 1]
```diff
@@ -153,6 +153,8 @@ func isEmptyValue(v reflect.Value) bool {
 		return v.Float() == 0
 	case reflect.Interface, reflect.Pointer:
 		return v.IsNil()
+	case reflect.Struct:
```
Comment (ndyakov) [inline commands.go:156]: If we consider `struct` as unsupported in https://github.com/redis/go-redis/blob/0df0f3c88785aa16c996089b68b731a4e4a2fdc0/internal/hscan/hscan.go#L21 , should we filter better for `time.Time` here ? @gh73962

Comment (gh73962) [reply commands.go:156]: code will be 
```go
    case reflect.Struct:
        if v.Type() == reflect.TypeOf(time.Time{}) {
            return v.IsZero()
        }
        // Only supports the struct time.Time, 
        // subsequent iterations will follow the func Scan support decoder.

```
What do u think about this change?

**Comment by gh73962** [comment]
> @ndyakov pls review and hset.Scan not support struct 图片

Additional note: hGetAll.Scan does not support parsing into nested structs, not that Scan doesn't support structs in general. The current issue is that hset supports implementing the BinaryMarshaler interface, while Scan fails to support nested BinaryUnmarshaler, creating an asymmetry in functionality. I believe redisv9's hset and scan are very developer-friendly, which is why my team migrated from v6 to v9. I hope go-redis can align the capabilities of hset and scan in the future. Thank you very much.
@ndyakov

**Comment by gh73962** [comment]
@ndyakov hi，pls check this PR

**Comment by gh73962** [comment]
@ndyakov pls review

**Review by ndyakov — APPROVED** [review]
Added one additional case, overall looks good.
